assets/src/edit-story/components/canvas/useInsertElement.js

```diff
@@ -45,12 +45,12 @@ function useInsertElement() {
    * is complete.
    */
   const backfillResource = useCallback(
-    (resource, elementId) => {
-      const { type, src, videoId, posterId } = resource;
+    (resource) => {
+      const { type, src, id, posterId } = resource;
 
       // Generate video poster if one not set.
-      if (type === 'video' && videoId && !posterId) {
-        uploadVideoPoster(videoId, src, elementId, resource);
+      if (type === 'video' && id && !posterId) {
```

Comment:
Think we should check that the resource isn't local, just in case.

Reply:
We already have 2 JIC checks: id is not set for local resource & we have this processing array.

Reply:
I guess a local video would not have an id, so would not pass. Just trying to be careful.